include/ignition/gui/GuiEvents.hh

```diff
@@ -323,6 +323,29 @@ namespace ignition
         /// \brief Private data pointer
         IGN_UTILS_IMPL_PTR(dataPtr)
       };
+
+      /// \brief Event which is called to broadcast the 2D coordinates of a
+      /// user's mouse hover within the scene.
+      class IGNITION_GUI_VISIBLE HoverOnScene : public QEvent
+      {
+        /// \brief Constructor
+        /// \param[in] _point The point at which the mouse is hovering within
+        /// the scene
+        public: explicit HoverOnScene(const math::Vector2i &_point);
```

Comment:
How about using `common::MouseEvent` like we're doing on `(Right|Left)ClickOnScene`?

Reply:
@chapulina 

The hover event in the QML is just providing these two values

https://github.com/ignitionrobotics/ign-gui/blob/6f50c9c2212705f83a78591de9f5943a54eca11f/src/plugins/minimal_scene/MinimalScene.qml#L42-L44

Another option is to extend the current `common::MouseEvent` to allow hover events:

https://github.com/ignitionrobotics/ign-common/blob/491d95588d0f5bcea4d4b3d4ee08b7a2ef982cb2/events/include/ignition/common/MouseEvent.hh#L54-L70

or `MOVE` is the same as hover ?

Reply:
Yeah `MOVE` with false `Dragging` is a hover :wink:

Reply:
I think the main advantage of `MouseEvent` is that we can add modifier keys to the event later as needed. For example, hover + Shift